src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseReport.java

```diff
@@ -0,0 +1,603 @@
+      // quick and dirty check for Slides
+      Matcher m = Pattern.compile(".*(\\d{1,5}-\\d{1,5}).*").matcher(dto.getName());
+      if (!m.matches()) throw new IllegalArgumentException("Sample with alias " + dto.getName() + " is malformed.");
+      timesReceived = m.group(1).split("-")[1];
+    } else {
+      return false;
+    }
+    return Integer.valueOf(barcodeString) != Integer.valueOf(timesReceived);
+  };
+
+  private final Predicate<SampleDto> byTimePointCodingMismatch = dto -> {
+    // finds slides that are not coded as biopsy (6) or radical prostatectomy (7) in the sample alias
+    if (getAttribute(ATTR_SLIDES, dto) == null) return false; // quick and dirty check for slides
+    Matcher m = Pattern.compile(".*(\\d{1,5}-\\d{1,5}).*").matcher(dto.getName());
+    if (!m.matches()) throw new IllegalArgumentException("Sample with alias " + dto.getName() + " is malformed.");
+    String timePoint = m.group(1).split("-")[0];
+    return Integer.valueOf(timePoint) < 6;
+  };
+
+  private final Predicate<SampleDto> byNotEmpty = sample -> {
+    if (sample.getSampleType() == null)
+      throw new IllegalArgumentException(String.format("Got very weird sample type for sample %s", sample.getName()));
+    if (sample.getSampleType().contains("Illumina") || "Unknown".equals(sample.getSampleType())) return false;
+    switch (sample.getSampleType()) {
+    case SAMPLE_CLASS_SLIDE:
+      // slides remaining
+      Integer slides = getIntAttribute(ATTR_SLIDES, sample);
+      Integer discards = getIntAttribute(ATTR_DISCARDS, sample);
+      if (discards == null) {
+        return slides > 0;
+      }
+      return slides > discards;
+    default:
+      return sample.getStorageLocation() != null && sample.getStorageLocation() != "" && sample.getStorageLocation() != "EMPTY";
+    }
+  };
+
+  @SafeVarargs
+  private final List<List<String>> getSingleTable(List<SampleDto> possibleSamples, Map<String, SampleDto> allSamplesById,
+      List<SampleLabel> labelList, Predicate<SampleDto>... filters) {
+    List<Predicate<SampleDto>> inputFilters = Arrays.asList(filters);
+    List<SampleDto> filtered = filter(possibleSamples, inputFilters);
+    if (filtered.isEmpty()) {
+      // nothing to report on here, so skip it
+      return Collections.emptyList();
+    }
+
+    return SITE_ORDER.stream()
+        .map(site -> {
+          List<Predicate<SampleDto>> allFilters = new ArrayList<>();
+          allFilters.add(Site.predicate(site));
+          allFilters.addAll(inputFilters);
+
+          List<SampleDto> sams = filter(possibleSamples, allFilters);
+          List<String> row = new ArrayList<>();
+          row.add(site.getKey());
+          row.addAll(getCounts(sams, allSamplesById, labelList));
+          return row;
+        })
+        .collect(Collectors.toList());
+  }
+
+  private List<String> getCounts(List<SampleDto> samples, Map<String, SampleDto> samplesById, List<SampleLabel> labels) {
```

Comment:
Please comment to describe returned data as it's a bit confusing. Returns the sample count and identity count per label in the same order as the `labels` list?